### PR TITLE
add new network session struct which wraps urlsession

### PR DESCRIPTION
**Pull request by ndurell**
* not hooked up at all

Not too much testing done yet. I plan to try to load it up in the test app before merging but technically nothing has changing with the existing url session.

NOTE: I wanted to hook this up to the old code but got some warning and felt it was safer to leave it alone for now.

### CODE REVIEW

File: Sources/KlaviyoSwift/Klaviyo.swift [hunk 1]
```diff
@@ -91,11 +93,11 @@ public class Klaviyo : NSObject {
     var apiKey : String?
     var apnDeviceToken : String?
     var userEmail : String = ""
-    var serialQueue : DispatchQueue!
```
Comment (ndurell) [inline Sources/KlaviyoSwift/Klaviyo.swift:94]: This is just random cleanup...

File: Sources/KlaviyoSwift/Klaviyo.swift [hunk 2]
```diff
@@ -112,22 +114,20 @@ public class Klaviyo : NSObject {
     Singleton Initializer. Must be kept private as only one instance can be created.
     */
     private override init() {
-        super.init()
-        
-        // Dev warnings for incorrect iOS version
-        assert(isOperatingMinimumiOSVersion() == true, "operating outdated iOS version. requires >= ios 13")
```
Comment (ndurell) [inline Sources/KlaviyoSwift/Klaviyo.swift:118]: More cleanup I don't think this ever did anything... :(

File: Tests/KlaviyoSwiftTests/KlaviyoTestUtils.swift [hunk 3]
```diff
@@ -35,7 +35,8 @@ extension KlaviyoEnvironment {
         archiverClient: ArchiverClient.test,
         fileClient: FileClient.test,
         data: { _ in TEST_RETURN_DATA },
-        logger: LoggerClient.test
+        logger: LoggerClient.test,
+        networkSession: NetworkSession.production
```
Comment (ndurell) [inline Tests/KlaviyoSwiftTests/KlaviyoTestUtils.swift:39]: Going to make the test and failure client in a future PR.

File: Sources/KlaviyoSwift/Klaviyo.swift [hunk 4]
```diff
@@ -112,22 +113,20 @@ public class Klaviyo : NSObject {
...
         let config = URLSessionConfiguration.default
         config.allowsCellularAccess = true
         config.httpMaximumConnectionsPerHost = urlSessionMaxConnection
         urlSession = URLSession(configuration: config)
-        reachability = Reachability(hostname: "a.klaviyo.com")
+        serialQueue = DispatchQueue(label: "com.klaviyo.serialQueue")
+        reachability = Reachability(hostname: "a.klaviyo.com")!
+        
+        super.init()
+        // Create the queue
```
Comment (kennyklaviyo) [inline Sources/KlaviyoSwift/Klaviyo.swift:125]: trailing comment?

Comment (ndurell) [inline Sources/KlaviyoSwift/Klaviyo.swift:125]:
```suggestion
        // Create the queue
        serialQueue = DispatchQueue(label: "com.klaviyo.serialQueue")
        reachability = Reachability(hostname: "a.klaviyo.com")!
        
        super.init()
```

File: Sources/KlaviyoSwift/Klaviyo.swift [hunk 5]
```diff
@@ -112,22 +113,20 @@ public class Klaviyo : NSObject {
...
-        reachability = Reachability(hostname: "a.klaviyo.com")
+        // Create the queue
+        serialQueue = DispatchQueue(label: "com.klaviyo.serialQueue")
+        reachability = Reachability(hostname: "a.klaviyo.com")!
+        
+        super.init()
+
+
+
+
```
Comment (kennyklaviyo) [inline Sources/KlaviyoSwift/Klaviyo.swift:129]: do we need all these lines?

Comment (ndurell) [reply Sources/KlaviyoSwift/Klaviyo.swift:129]: Oops

Comment (ndurell) [inline Sources/KlaviyoSwift/Klaviyo.swift:129]:
```suggestion
```

File: Sources/KlaviyoSwift/NetworkSession.swift [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        let version = ProcessInfo.processInfo.operatingSystemVersion
+        let versionString = "\(version.majorVersion).\(version.minorVersion).\(version.patchVersion)"
+        let osName: String = "iOS"
+
+        return "\(osName) \(versionString)"
+    }()
+
+    let klaivyoSDKVersion = "klaviyo-ios/\(version)"
+
+    return "\(executable)/\(appVersion) (\(bundle); build:\(appBuild); \(osNameVersion)) \(klaivyoSDKVersion)"
```
Comment (kennyklaviyo) [inline Sources/KlaviyoSwift/NetworkSession.swift:29]: can we easily see % by sdk version this way? or should be have individual headers for each data point

Comment (ndurell) [reply Sources/KlaviyoSwift/NetworkSession.swift:29]: In splunk it should show up.
Happy to split it up too...we hadn't decided on names yet. Also I was planning on potentially including a lot of the same  in profile updates and events.

**Review by kennyklaviyo — APPROVED** [review]

**Pull request merged**